app/api/answers/generate_task_token.feature

```diff
@@ -0,0 +1,251 @@
+      | 104      | 20      | content            | answer              |
+      | 104      | 30      | content            | answer              |
+    And the database has the following table 'attempts':
+      | id | participant_id |
+      | 0  | 101            |
+      | 0  | 102            |
+      | 0  | 103            |
+      | 1  | 103            |
+    And the database has the following table 'results':
+      | attempt_id | participant_id | item_id | started_at          | validated_at        |
+      | 0          | 101            | 10      | 2020-01-01 01:01:01 | null                |
+      | 0          | 101            | 20      | 2020-01-01 01:01:01 | null                |
+      | 0          | 101            | 30      | 2020-01-01 01:01:01 | null                |
+      | 0          | 102            | 20      | 2020-01-01 01:01:01 | null                |
+      | 0          | 103            | 10      | 2020-01-01 01:01:01 | null                |
+      | 0          | 103            | 20      | 2020-01-01 01:01:01 | null                |
+      | 0          | 103            | 30      | null                | null                |
+      | 1          | 103            | 30      | 2020-01-01 01:01:01 | null                |
+      | 0          | 101            | 40      | 2020-01-01 01:01:01 | null                |
+      | 0          | 101            | 50      | 2020-01-01 01:01:01 | 2020-01-01 01:01:01 |
+    And the database has the following table 'answers':
+      | id | participant_id | attempt_id | item_id | author_id  | created_at          |
+      | 1  | 101            | 0          | 10      | 101        | 2020-01-01 01:01:01 |
+      | 2  | 102            | 0          | 20      | 101        | 2020-01-01 01:01:01 |
+      | 3  | 101            | 0          | 30      | 101        | 2020-01-01 01:01:01 |
+      | 4  | 101            | 0          | 40      | 101        | 2020-01-01 01:01:01 |
+      | 5  | 101            | 0          | 50      | 101        | 2020-01-01 01:01:01 |
+    And time is frozen
+
+  Scenario: User is able to fetch a task token when participant is the current user
+    Given I am the user with id "101"
+    When I send a POST request to "/answers/1/generate-task-token"
+    Then the response code should be 200
+    And the response body decoded as "GenerateTaskTokenResponse" should be, in JSON:
+      """
+      {
+        "data": {
+          "task_token": {
+            "date": "{{currentTimeInFormat("02-01-2006")}}",
+            "bAccessSolutions": false,
+            "bHintsAllowed": false,
+            "bIsAdmin": false,
+            "bReadAnswers": true,
+            "bSubmissionPossible": false,
+            "idAttempt": "101/0",
+            "idUser": "101",
+            "idItemLocal": "10",
+            "idItem": "task10",
+            "itemUrl": "http://taskplatform.mblockelet.info/task.html?taskId=403449543672183936",
+            "nbHintsGiven": "0",
```

Comment:
`nbHintsGiven` should be tested as well, it is always at "0" in the tests

Reply:
fixed